apps/prairielearn/python/python_helper_sympy_test.py

```diff
@@ -317,12 +317,12 @@ def test_invalid_function(self, a_sub: str) -> None:
 
     @pytest.mark.parametrize("a_sub", INVALID_VARIABLE_CASES)
     def test_invalid_variable(self, a_sub: str) -> None:
-        with pytest.raises(phs.HasInvalidSymbolError):
+        with pytest.raises((phs.HasInvalidSymbolError, phs.HasInvalidVariableError)):
```

Comment:
I'm confused about the difference between `HasInvalidSymbolError` and `HasInvalidVariableError` – I accidentally changed this to InvalidVariable, and then the test started failing. That's why I changed the test case to `exp(y)` above, because I thought that might hit the variable check code, but no. When can the `HasInvalidVariableError` code actually trigger?

Reply:
I think the invalid variable error is pretty hard to trigger, it might be worth playing around to see if it can be applied in more cases. Maybe if the CheckVariables call were done before the CheckFunctions call? This structure was present in the old code and I kept it just to avoid potentially causing more issues, but it might be worth scrapping to make the code less confusing.